scr/script.js

```diff
@@ -0,0 +1,105 @@
+if ('library' in localStorage && localStorage.getItem('library') === '') {
+  const myLibrary = [];
+  const i = 0;
+  localStorage.setItem('counter', i);
+  localStorage.setItem('library', myLibrary);
+}
+
+function Book(name, author, pages, read, num) {
+  this.name = name;
+  this.author = author;
+  this.pages = pages;
+  this.read = read;
+  this.num = num;
+}
+
+function addBookToLibrary(name, author, pages, rd1 = false, rd2 = false) {
+  let myLibrary = [];
+  let read;
+  if ('library' in localStorage && localStorage.getItem('library') !== '') {
+    myLibrary = JSON.parse(localStorage.getItem('library'));
+  }
+
+  let i = JSON.parse(localStorage.getItem('counter'));
+  const num = i;
+
+  if (rd1.checked === true) {
+    read = rd1.value;
+  } else if (rd2.checked === true) {
+    read = rd2.value;
+  } else {
+    read = 'unspecified';
+  }
+
+  myLibrary[i] = new Book(name, author, pages, read, num);
+  i += 1;
+  localStorage.setItem('library', JSON.stringify(myLibrary));
+  localStorage.setItem('counter', JSON.stringify(i));
+}
+
+function forma() {
+  document.getElementById('form').style.display = 'block';
+}
+
+function cancel() {
+  document.getElementById('form').style.display = 'none';
+}
+
+function cont() {
+  let myLibrary = JSON.parse(localStorage.getItem('library'));
```

Comment:
this should be an immutable variable since it's coming from local storage, try using destructuring in order to access does variables. By fixing this issue you will find the answer of why your code is braking